### PR TITLE
feat(*) add service mesh patches

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Adds OpenResty and Nginx Patches for Service Mesh support.

#### Instructions

```
wget https://openresty.org/download/openresty-1.13.6.2.tar.gz
tar zxvf openresty-1.13.6.2.tar.gz
wget https://github.com/Kong/openresty-patches/archive/feat/service-mesh-patches.tar.gz
tar zxvf service-mesh-patches.tar.gz
cd openresty-1.13.6.2/bundle/
for i in ../../openresty-patches-feat-service-mesh-patches/patches/1.13.6.2/*.patch; \
  do patch -p1 < $i; done
```

Then build and try!

#### See Also

https://github.com/Kong/openresty-patches/blob/2893c829a0466f37e52bbcf790f71a13b4c783f7/README.md

### CODE REVIEW

**Comment by james-callahan** [comment]
Need to sit down and look at this harder; not an easy review!

Would it be possible to split this up into multiple pull requests? Or do you think that doesn't make sense?

Do we run any tests before this gets merged?

**Comment by bungle** [comment]
When this gets merged, the travis will start to apply these patches automatically, so Kong test suite will be running with these patches applied. I think you can do that too on your own machine.

Things to review:
1. every patch needed is added
2. there is no patches that we don't want
3. there are no patches that conflict?
4. your private service mesh branch works with OpenResty built with these patches
5. OpenResty test suite works?
6. Kong test suite works with OpenResty build with these patches?

Yes, it is hard to review, but these patches come mainly from you, so I think you are the only one to really verify these. Kong should still work even without these patches, it is only that service mesh won't (that might change in a future).

There are a lot of patches in `master` branches of respective projects, so it is always a bit worrying to backport these to lower versions, without backporting them all, as you never know how they will apply, or break anything. E.g. I needed to add semaphore patches because Yichun's patch was very hard to apply without it.

File: patches/1.13.6.2/nginx-1.13.6_02-listen-transparent.patch [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
++                    {
++                        ngx_log_error(NGX_LOG_EMERG, log, ngx_socket_errno,
++                                      "setsockopt(IP_TRANSPARENT) %V failed",
++                                      &ls[i].addr_text);
++                        return NGX_ERROR;
++                    }
++
++                    break;
++
++#if (NGX_HAVE_INET6)
```
Comment (james-callahan) [inline patches/1.13.6.2/nginx-1.13.6_02-listen-transparent.patch]: May need to add `defined(IPV6_TRANSPARENT)` here. I recall running into a system without it when testing my patch further.

**Comment by bungle** [comment]
@james-callahan ^ it is now split if you want to go and review then piece by piece. I will try your service mesh talk demos tomorrow with all these patches applied, and see if I can get it running.

**Review by james-callahan — APPROVED** [review]

File: patches/1.13.6.2/nginx-1.13.6_02-listen-transparent.patch (hunk 1 above)
Comment (bungle) [reply patches/1.13.6.2/nginx-1.13.6_02-listen-transparent.patch]: @james-callahan,

So, do you want me to change the line:

```c
#if (NGX_HAVE_INET6)
```

to

```c
#if (NGX_HAVE_INET6 && defined(IPV6_TRANSPARENT))
```

**Pull request merged**